torch_geometric/nn/aggr/patch_transformer.py

```diff
@@ -0,0 +1,159 @@
+from torch import Tensor
+
+from torch_geometric.experimental import disable_dynamic_shapes
+from torch_geometric.nn.aggr import Aggregation
+from torch_geometric.nn.aggr.pt_utils import (
+    FeatEncoder,
+    PositionalEncoding1D,
+    TransformerBlock,
+)
+
+
+class PatchTransformerAggregation(Aggregation):
+    r"""Performs  Heterogeneous Link Encoder embedding
+    and Semantic Patches Fusion
+    from `"Simplifying Temporal Heterogeneous Network for
+    Continuous-Time Link Prediction
+    "<https://dl.acm.org/doi/pdf/10.1145/3583780.3615059>`_paper.
+
+    Args:
+        channels (int): Size of each edge feature
+        use to specify the time embedding dimension of edge timestamps
+        hidden_dim (int, optional): hidden dimension
+        out_dim (int, optional): output dimension
+        max_edge (int, optional): max number of edges per node
+        num_encoder_blocks (int, optional): Number of Set Attention Blocks
+            (SABs) in the encoder. (default: :obj:`1`).
+        heads (int, optional): Number of multi-head-attentions.
+            (default: :obj:`1`)
+        patch_size (int, optional): Number of links in a patch
+            (default: :obj:`5`)
+        norm (str, optional): If set to :obj:`True`, will apply layer
+            normalization. (default: :obj:`False`)
+        dropout (float, optional): Dropout probability of attention weights.
+            (default: :obj:`0`)
+        aggr (list[str], optional): aggregation module, ['sum','max','mean']
+        time_dim (int, optional): time embedding dimension,
+
+    """
+    def __init__(
+        self,
+        channels: int,
+        hidden_dim: int = 64,
+        out_dim: int = 128,
+        max_edge: int = 16,
+        num_encoder_blocks: int = 1,
+        heads: int = 1,
+        patch_size: int = 5,
+        layer_norm: bool = False,
+        dropout: float = 0.0,
+        aggr: list[str] = None,
+        time_dim: int = 0,
+    ):
+        super().__init__()
+        self.channels = channels
+        self.time_dim = time_dim
+        self.hidden_dim = hidden_dim
+        self.out_dim = out_dim
+        self.heads = heads
+        self.layer_norm = layer_norm
+        self.patch_size = patch_size
+        self.dropout = dropout
+        self.max_edge = max_edge
+        self.aggr = aggr
+        if (self.aggr is None):
+            self.aggr = ["mean"]
+
+        # encoders and linear layers
+        self.feat_encoder = FeatEncoder(self.channels, self.hidden_dim,
+                                        time_dim=self.time_dim)
+        self.layernorm = torch.nn.LayerNorm(self.hidden_dim)
+        self.atten_blocks = torch.nn.ModuleList([
+            TransformerBlock(dims=self.hidden_dim, heads=self.heads,
+                             dropout=self.dropout)
+            for _ in range(num_encoder_blocks)
+        ])
+
+        # input is node feature and transformer node embed
+        self.mlp_head = torch.nn.Linear(self.hidden_dim * (len(self.aggr)),
```

Comment:
As this is just one layer, maybe something like self.lin or self.fc is a better name?

Reply:
addressed in recent commit